common/src/main/java/org/broadleafcommerce/common/resource/service/ResourceMinificationServiceImpl.java

```diff
@@ -64,20 +80,45 @@
     @Value("${minify.disableOptimizations}")
     protected boolean disableOptimizations;
 
+    @Value("${minify.closure.compiler.enabled}")
+    protected boolean closureCompilerEnabled;
```

Comment:
Having a boolean to tell a service to do one of two very different implementations is a dead giveaway that we need to have 2 implementations of `ResourceMinificationService`. One that implements how we do it for YUICompressor and one for Google Closure Compiler. We can either create the bean `blResourceMinificationService` in a configuration file based on if Google Closure Compiler is on the classpath or not or we can add that `@Conditional` directly on the class an let component scanning dictate which one is created. If there's common logic between the two minification service then an Abstract class should be created that has the common logic implemented